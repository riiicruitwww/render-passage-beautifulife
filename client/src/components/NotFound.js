import React from 'react';
import { Link } from 'react-router-dom';
import './NotFound.scss';

const NotFound = () => {
  return (
    <div className="NotFound">
      <h1>존재하지 않는 페이지 입니다.</h1>
      <h2>404 Not Found</h2>
      <Link to="/">메인 페이지로</Link>
    </div>
  );
};

export default NotFound;
