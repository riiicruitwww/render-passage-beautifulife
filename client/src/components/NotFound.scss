.NotFound {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  color: white;

  a {
    padding: 0.5em;
    border: 1px solid;
    border-radius: 20px;
  }
}
