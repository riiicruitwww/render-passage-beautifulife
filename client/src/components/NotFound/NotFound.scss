.NotFound {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 192px;
  height: 100vh;
  color: #ffffff;

  a {
    padding: 8px;
    border: 1px solid;
    border-radius: 20px;
    color: #e0e0e0;
    opacity: 0.7;
    transition: all 0.3s;
    cursor: pointer;

    &:hover {
      opacity: 1;
    }
  }
}
