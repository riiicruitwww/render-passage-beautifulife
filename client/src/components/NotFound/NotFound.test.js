import React from 'react';
import { shallow } from 'enzyme';
import NotFound from './NotFound';

describe('NotFound', () => {
  describe('about render', () => {
    const wrapper = shallow(<NotFound />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
