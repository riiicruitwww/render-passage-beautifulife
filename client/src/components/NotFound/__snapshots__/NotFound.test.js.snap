// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          존재하지 않는 페이지 입니다.
        </h1>,
        <h2>
          404 Not Found
        </h2>,
        <Link
          to="/"
        >
          메인 페이지로
        </Link>,
      ],
      "className": "NotFound",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "존재하지 않는 페이지 입니다.",
        },
        "ref": null,
        "rendered": "존재하지 않는 페이지 입니다.",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404 Not Found",
        },
        "ref": null,
        "rendered": "404 Not Found",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "메인 페이지로",
          "to": "/",
        },
        "ref": null,
        "rendered": "메인 페이지로",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            존재하지 않는 페이지 입니다.
          </h1>,
          <h2>
            404 Not Found
          </h2>,
          <Link
            to="/"
          >
            메인 페이지로
          </Link>,
        ],
        "className": "NotFound",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "존재하지 않는 페이지 입니다.",
          },
          "ref": null,
          "rendered": "존재하지 않는 페이지 입니다.",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404 Not Found",
          },
          "ref": null,
          "rendered": "404 Not Found",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "메인 페이지로",
            "to": "/",
          },
          "ref": null,
          "rendered": "메인 페이지로",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
